File: Leetcode/2293. Min Max Game.cpp
class Solution {
private:
    int helper(vector <int> &nums, int N) {
        if (N == 1) return nums[0];
        else {
            for (int i = 0; i < N / 2; i++) {
                nums[i] = (i % 2) ? max(nums[2 * i], nums[2 * i + 1]) : min(nums[2 * i], nums[2 * i + 1]);
            }
            return helper(nums, N / 2);
        }
    }
public:
    int minMaxGame(vector<int>& nums) {
        return helper(nums, nums.size());
    }
};

File: Leetcode/2293. Min Max Game.py
class Solution:
    def minMaxGame(self, nums: List[int]) -> int:
        def helper(nums, N):
            if N == 1: return nums[0]
            else:
                for i in range(N // 2):
                    nums[i] = max(nums[2 * i], nums[2 * i + 1])  if (i % 2 == 1) else min(nums[2 * i], nums[2 * i + 1])
                return helper(nums, N // 2);
        return helper(nums, len(nums))